lib/rules/indent.js

```diff
@@ -896,7 +896,11 @@ module.exports = {
 
             parameterParens.add(openingParen);
             parameterParens.add(closingParen);
-            offsets.matchIndentOf(sourceCode.getTokenBefore(openingParen), openingParen);
+
+            // https://github.com/eslint/eslint/issues/8815
+            const tokenBefore = sourceCode.getTokenBefore(openingParen, node.parent.type === "AwaitExpression" ? astUtils.isKeywordToken : void 0);
```

Comment:
I don't think this fix is quite correct. I think the problem is actually with the logic for `AssignmentExpression` and `VariableDeclaration` nodes, not with `CallExpression` nodes -- the from the report in https://github.com/eslint/eslint/issues/8815 is just a special case that happens to include `AwaitExpression` and a `CallExpression`.

For example, the following code is also an error due to this bug, but it's not fixed:

```js
const {
    foo
} = typeof foo(
    1
);
```

Instead of modifying the logic for `CallExpression`s, I think the following fix solves the problem:

```diff
diff --git a/lib/rules/indent.js b/lib/rules/indent.js
index 6babe29c..fafd1292 100644
--- a/lib/rules/indent.js
+++ b/lib/rules/indent.js
@@ -1030,7 +1030,7 @@ module.exports = {
                 const nodeTokens = getTokensAndComments(node);
                 const tokensFromOperator = nodeTokens.slice(lodash.sortedIndexBy(nodeTokens, operator, token => token.range[0]));

-                offsets.setDesiredOffsets(tokensFromOperator, sourceCode.getFirstToken(node.left), 1);
+                offsets.setDesiredOffsets(tokensFromOperator, sourceCode.getLastToken(node.left), 1);
                 offsets.ignoreToken(tokensFromOperator[0]);
                 offsets.ignoreToken(tokensFromOperator[1]);
             },
@@ -1325,12 +1325,12 @@ module.exports = {
             VariableDeclarator(node) {
                 if (node.init) {
                     const equalOperator = sourceCode.getTokenBefore(node.init, astUtils.isNotOpeningParenToken);
-                    const tokenAfterOperator = sourceCode.getTokenAfter(equalOperator);
+                    const tokensAfterOperator = sourceCode.getTokensAfter(equalOperator, token => token.range[1] <= node.range[1]);

                     offsets.ignoreToken(equalOperator);
-                    offsets.ignoreToken(tokenAfterOperator);
-                    offsets.matchIndentOf(equalOperator, tokenAfterOperator);
-                    offsets.matchIndentOf(sourceCode.getFirstToken(node), equalOperator);
+                    offsets.ignoreToken(tokensAfterOperator[0]);
+                    offsets.setDesiredOffsets(tokensAfterOperator, equalOperator, 1);
+                    offsets.matchIndentOf(sourceCode.getLastToken(node.id), equalOperator);
                 }
             },

```

There are two parts to the fix:

1. With `AssignmentExpression` nodes, the equals sign should be offset from the last token of the `AssignmentExpression`, not the first token. This prevents token-collapsing bugs like [this one](https://github.com/eslint/eslint/blob/5c3ac8ed3c1d8f4b8f4cd82d7c5d79d88de26ec5/lib/rules/indent.js#L270-L291).
1. With `VariableDeclarator` nodes, all tokens should be indented from the `=` token, not just the first token after the `=` token. The problem before was that in a case like this:

    ```js
    var {
    } = await foo(
    )
    ```

    Only the `await` token's indentation was getting set to be aligned with the `}`, but the `foo` token's indentation was not getting aligned properly, so the `foo()` expression was offset from `}` when it should have been aligned instead.